lua/telescope/_extensions/live_grep_args.lua

```diff
@@ -51,6 +51,11 @@ local live_grep_args = function(opts)
     return cmd
   end
 
+  -- apply theme
+  if opts.theme then
```

Comment:
Seems like this will mean only being able to specify the theme as a string rather than a user using their own custom theme which would be a table. I contributed a fix for a similar issue in dressing.nvim a while ago, which looked like https://github.com/stevearc/dressing.nvim/pull/20/files

But eventually the author came up with a better solution which was just to pass the value a user passes directly to the picker, something like https://github.com/stevearc/dressing.nvim/blob/master/lua/dressing/select/telescope.lua#L82-L84.

Note that it in the case of dressing a user has to pass the theme themselves e.g.
```lua
-- Options for telescope selector
    -- These are passed into the telescope picker directly. Can be used like:
    -- telescope = require('telescope.themes').get_ivy({...})
    telescope = nil,
```
It might be slightly more verbose but is much more flexible than having to only use a `string`

Reply:
@akinsho some points to take here.

I may be wrong but it seems that dressing is setting `dropdown` as default theme, and if picker have custom options they will be passed on. This works fine for `builtin` pickers, but doesn't work out-of-the-box for extensions. As you mentioned, you have to wrap the opts with a theme from `telescope.themes` or your own theme.

I know you can create a custom theme with the options `layout_config = {}`, `border = bool`, `borderchars = {...}`, and others, I even do this my self here. But not sure about theme as table like this `theme = {...}`. Don't think it is a thing in telescope, i mean, `builtin` pickers only accept string for the `theme` option.

What i do here is to apply themes the same whay telescope does in `apply_config` for the `builtin` pickers:

https://github.com/nvim-telescope/telescope.nvim/blob/master/lua/telescope/builtin/init.lua#L485-L487

If a theme string is provided theme will be set using one of the `telescope.themes`, otherwise you are still able to use your custom theme with `borderchars` etc, by wraping the options when calling it:

```
:lua require("telescope").extensions.live_grep_args.live_grep_args(my_custom_theme(opts))
```

Or using telescope themes:
```
:lua require("telescope").extensions.live_grep_args.live_grep_args(require('telescope.themes').get_ivy(opts))
```

Reply:
@akinsho sorry but I'm not getting your point.

Let me try. All you want to do is this right? Let us say that you have your custom theme function:

```lua
function my_custom_theme(opts)
    custom_opts = {
        -- my custom borders for example
    }
    return ...
end
```

Instead of passing the custom theme like this:
```
:lua require("telescope").extensions.live_grep_args.live_grep_args(my_custom_theme(opts))
```

You want that `live_grep_args` supports the use of this theme function like this in the config:

```lua
telescope.setup {
  extensions = {
    live_grep_args = my_custom_theme({
      auto_quoting = true,
    })
    -- live_grep_args = my_custom_theme() -- with defaults
  }
}
```
And then call `live_grep_args` like this:
```
:lua require("telescope").extensions.live_grep_args.live_grep_args()
```
Is that it? If that is your point I don't get it, because this is already possible, even before my change.

I may be wrong but the entire opts is being passed to the picker creation in the same way as your example, and as in builtin pickers. The only thing I do is basically to add support to use builtin themes passing the theme name in the `theme` option just like the builtin pickers, but other customization works just like any other picker, for instance using a custom theme function like this `live_grep_args = my_custom_theme()`, or passing pickers borders directly in the opts for example, this works just like builtin pickers, or at least it should, if its not working let us know.

If that is not your point I really don't get it.

Reply:
Now If i'm entirely wrong about everything and what you want is to use the `theme` option not only to set a builtin theme with the theme name, but also to allow it to receive a custom theme as a table (containing the custom borders for instance). 

If that is the case, I can see more or less how this could work, but i would prefer not to do that, as implementing a way to setup themes different then the builtin's way seems out of the scope of `live_grep_args` extension.

Reply:
I will resolve this conversation. Even if I unable to follow it completely.

The config options work, probably with their limits. We can improve on this later.